### PR TITLE
Use hyperchain contract from external repository

### PR DESCRIPTION
**Pull request by rustaem**
Addressing #3679.

### CODE REVIEW

**Comment by cytadela8** [comment]
I believe this replaces #3609?

**Comment by rustaem** [comment]
> I believe this replaces #3609?

Yes.

**Comment by cytadela8** [comment]
Rebase please, cause I rebased hyperchain-consensus branch.

**Comment by cytadela8** [comment]
Still needs a rebase

File: rebar.config [hunk 1]
```diff
@@ -28,86 +28,90 @@
...
-        {aesophia_aci_encoder, {git, "https://github.com/aeternity/aesophia_aci_encoder",
-          {ref, "41d0427"}}}
-
-       ]}.
+    %% off-the-shelf depdendencies
+	{enoise, {git, "https://github.com/aeternity/enoise.git", {ref, "1e6ee67"}}},
+	{gproc, "0.6.1"},
+	{jobs, "0.9.0"},
+	{exometer_core, "1.5.7"},
+    {yamerl, "0.7.0"},
```
Comment (cytadela8) [inline rebar.config:36]: Sorry for pointing out stupid issues, but if you want to improve the style make sure new one is consistant ;).

This line has wrong ident.

File: rebar.config [hunk 2]
```diff
@@ -28,86 +28,90 @@
...
+    % for hyperchains and the testing suite
+    {aesophia_aci_encoder, {git, "https://github.com/aeternity/aesophia_aci_encoder",
+        {ref, "41d0427"}}},
+
+    % For automatic building of the StakingContract.json
+    % Do NOT use 'tag' or 'ref' clause here as the source code of the contract is subject to change.
+    {hyperchains_contract, {git, "git://github.com/aeternity/hyperchains-contract.git",
+        {branch, "master"}}}
+
+%%    And this here is for testing purposes when you don't want to merge both PRs at once.
```
Comment (cytadela8) [inline rebar.config:111]: Remove as that PR is merged

**Comment by cytadela8** [comment]
We are gonna merge this soon. Please rebase.

File: Makefile [hunk 3]
```diff
@@ -410,13 +410,17 @@ multi-build: VERSION dev1-build
 # Build rules
 #
 
+<<<<<<< HEAD
```
Comment (cytadela8) [inline Makefile:413]: @rustaem ...

**Review by cytadela8 — CHANGES_REQUESTED** [review]

**Comment by cytadela8** [comment]
Helpfull resources:
- https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts
- https://medium.com/fredwong-it/git-rebase-how-to-use-interactive-rebase-properly-34db370be995
- VIDEO https://www.simplilearn.com/tutorials/git-tutorial/merge-conflicts-in-git

File: Makefile (hunk 3 above)
Comment (rustaem) [reply Makefile]: > @rustaem ...

Sorry. Never should have make myself too accustomed to GUI process :(

File: Makefile [hunk 4]
```diff
@@ -415,13 +412,10 @@ multi-build: VERSION dev1-build
 # Build rules
 #
 
-internal-compile-deps: hc-compile-staking-contract
-	@$(REBAR) as $(KIND) compile -d
```
Comment (cytadela8) [inline Makefile:419]: Is this not needed?

**Review by cytadela8 — COMMENTED** [review]
The tests are failing. Please check why and fix. One of them failed with: `Hyperchains staking contract not present. Terminating the node - file not found "/home/builder/aeternity/data/aehyperchains/StakingContract.json"Hyperchains staking contract not present.`

**Comment by cytadela8** [comment]
Well, tests are still failing.

**Comment by rustaem** [comment]
> Well, tests are still failing.

Dia's fine, digging eunits.

**Review by cytadela8 — CHANGES_REQUESTED** [review]
Fixes are needed for how we compile and look for the contract. Found issues:
- hyperchain_contract uses a shell hook that assumes a system wide rebar3 is availble. This needs not be a case. We use the rebar3 from repository. Consider switching to provider hook from shell hook. https://rebar3.readme.io/docs/configuration#provider-hooks
- remove `ADD` line Dockerfile that attempts to add contract code
- fix how we look for the .json comiled contract code. Tests show that is is expected in the $projectroot/data/aehyperchains this worked with previous way of compiling, but new one only provides it in data in _build.

**Comment by seanhinde** [comment]
Superseded by new HC design

**Pull request closed without merging**